app/models/policy.rb

```diff
@@ -81,6 +93,13 @@ def destroy_orphaned_business_objective
 
   private
 
+  def unassigned_minor_versions
+    stored_versions = profiles.pluck(:os_minor_version)
+    Host.os_minor_versions(hosts.pluck(:id)).reject do |version|
```

Comment:
No need to `pluck` the ids here. Doing it as suggested uses a nested subquery vs. 2 independent queries.

```suggestion
    Host.os_minor_versions(hosts).reject do |version|
```

```rb
[307] pry(main)> Host.os_minor_versions(Host.limit(10))
   (64.3ms)  SELECT DISTINCT system_profile->'operating_system'->'minor' FROM "inventory"."hosts" WHERE "inventory"."hosts"."id" IN (SELECT  "inventory"."hosts"."id" FROM "inventory"."hosts" LIMIT $1)  [["LIMIT", 10]]
=> [7, 9]
```